CoreMapView/src/au/gov/asd/tac/constellation/views/mapview/overlays/OverviewOverlay.java

```diff
@@ -162,27 +162,48 @@ public void setDimension(final ScreenPosition topLeft, final ScreenPosition bott
         public void draw() {
             renderer.noFill();
             renderer.strokeWeight(2);
-            renderer.stroke(highlightColor);
-
-            float viewportX = Math.min(Math.max(x, getX()), getX() + getWidth());
-            float viewportWidth = x < getX() ? Math.max(width - (getX() - x), 0)
-                    : x + width > getX() + getWidth() ? Math.max(width - ((x + width) - (getX() + getWidth())), 0)
-                            : width;
-
-            float viewportY = Math.min(Math.max(y, getY()), getY() + getHeight());
-            float viewportHeight = y < getY() ? Math.max(height - (getY() - y), 0)
-                    : y + height > getY() + getHeight() ? Math.max(height - ((y + height) - (getY() + getHeight())), 0)
-                            : height;
+            renderer.stroke(HIGHLIGHT_COLOUR);
+
+            final float viewportX = Math.min(Math.max(x, getX()), getX() + getWidth());
+            final float viewportY = Math.min(Math.max(y, getY()), getY() + getHeight());
+            
+            final float viewportWidth;
+            final float viewportHeight;
+            if(x < getX()){
+                viewportWidth = Math.max(width - (getX() - x), 0);
+            }else{
+                if(x + width > getX() + getWidth()){
+                    viewportWidth = Math.max(width - ((x + width) - (getX() + getWidth())), 0);
+                }else{
+                    viewportWidth = width;
+                }
+            }
+            if(y < getY()){
+                viewportHeight = Math.max(height - (getY() - y), 0);
+            }else{
+                if(y + height > getY() + getHeight()){
+                    viewportHeight = Math.max(height - ((y + height) - (getY() + getHeight())), 0);
+                }else{
+                    viewportHeight = height;
+                }
+            }
 
             renderer.rect(viewportX, viewportY, viewportWidth, viewportHeight);
         }
 
         public void update(final int zoom) {
             final Location newLocation = overviewMap.mapDisplay.getLocation(
                     viewport.x + viewport.width / 2, viewport.y + viewport.height / 2);
-            final int newZoomLevel = zoom > 0 ? map.getZoomLevel() - 1
-                    : zoom < 0 ? map.getZoomLevel() + 1
-                            : map.getZoomLevel();
+            final int newZoomLevel;
+            if(zoom > 0) {
+                newZoomLevel = map.getZoomLevel() - 1;
+            }else{
+                if(zoom < 0){
+                    newZoomLevel=map.getZoomLevel() + 1;
+                }else{
+                    newZoomLevel=map.getZoomLevel();
```

Comment:
similar here. This could potentially also become else if{}else{} rather than else{if{}else{}}